src/components/Link.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+
+type LinkVariantProps = React.HTMLProps<HTMLAnchorElement> & {
+  isActive: boolean;
+  variant?: "link";
+  href: string;
+  setIsActive: (newValue: boolean) => void;
+};
+type NavVariantProps = React.HTMLProps<HTMLLIElement> & {
+  isActive: boolean;
+  variant: "nav";
+  onClick: () => void;
+};
+type Props = LinkVariantProps | NavVariantProps;
+
+/**
+ * A reusable link component with animated underline when the link is active.
+ *
+ * Has two variants:
+ *  1. "link" (default): renders an <a> tag and calls a setIsActive callback
+ *     when the link is (un)focused and (un)hovered.
+ *  2. "nav": renders a <li> tag with an onClick callback. Has no inherent
+ *     hover/focus behaviour.
+ */
+const Link = (props: Props) => {
+  const { isActive, variant, children } = props;
+
+  const className = `
+    ${variant === "nav" ? "mx-4" : null}
+    cursor-pointer
+    relative 
+    
+    after:block
+    after:absolute
+    after:border
+    after:h-px
+    after:top-full
+    after:left-0
+    after:transition-all
+    after:duration-300 
+
+    ${
+      isActive
+        ? "after:right-0 after:opacity-1"
+        : "after:right-full after:opacity-0"
+    }`.trim();
+
+  const Tag = variant === "nav" ? "li" : "a";
```

Comment:
For wcag we should generally use natively interactive elements - `button` and `a`. I think I'd prefer we add `button` here as a child of the original `li` element.

Reply:
Yep, good shout. I got too stuck on not wanting to change it too much from what was there, but you're absolutely right. Will make that change, and the couple consequential bits below too.